web/modules/custom/hoeringsportal_dialogue/src/Helper/DialogueHelper.php

```diff
@@ -83,6 +89,31 @@ public function dialogueProposalPresave(EntityInterface $entity): void {
     }
   }
 
+  /**
+   * Implements update for dialogue comments.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $comment
+   *   The comment that is being updated.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function dialogueCommentUpdate(EntityInterface $comment): void {
+    if ($this::DIALOGUE_PROPOSAL_COMMENT_TYPE === $comment->bundle()) {
+      $children = [];
+      /** @var \Drupal\comment\Entity\Comment $comment */
+      if (!$comment->isPublished() && $comment->original->isPublished()) {
+        $this->messenger->addMessage($this->t("Comment @commentId and it's children have been unpublished.", ['@commentId' => $comment->id()]));
+        $this->getDialogueCommentChildren($comment, $children);
```

Comment:
Move `$children` to right before it's used:

```suggestion
      /** @var \Drupal\comment\Entity\Comment $comment */
      if (!$comment->isPublished() && $comment->original->isPublished()) {
        $this->messenger->addMessage($this->t("Comment @commentId and it's children have been unpublished.", ['@commentId' => $comment->id()]));
        $children = [];
        $this->getDialogueCommentChildren($comment, $children);
```